docs/pages/9 - Contrib Modules.md

```diff
@@ -272,36 +272,252 @@ There's an [example project](https://github.com/lihaoyi/cask/tree/master/example
 
 ### Play Framework
 
-Play framework routes generation support.
+This module adds basic Play Framework support to mill: 
+- configures mill for Play default directory layout,
+- integrates the Play routes compiler,
+- optionally: integrates the Twirl template engine,
+- optionally: configures mill for single module play applications.
 
- 
-To declare a module that needs to generate Play Framework routes, you must mix-in the 
-`mill.playlib.routesModule` trait when defining your module. 
+There is no specific Play Java support, building a Play Java application will require a bit 
+of customization (mostly adding the proper dependencies).  
+
+#### Using the plugin
+
+There are 2 base modules and 2 helper traits in this plugin, all of which can be found
+ in `mill.playlib`.
+
+The base modules:  
+- `PlayModule` applies the default Play configuration (layout, dependencies, routes compilation, 
+Twirl compilation and Akka HTTP server)
+- `PlayApiModule` applies the default Play configuration without `Twirl` templating. This is useful 
+if your Play app is a pure API server or if you want to use a different templating engine.
+
+The two helper traits: 
+- `SingleModule` can be useful to configure mill for a single module Play application such as the 
+[play-scala-seed project](https://github.com/playframework/play-scala-seed.g8). Mill is 
+multi-module by default and requires a bit more configuration to have source, resource, and test 
+directories at the top level alongside the `build.sc` file. This trait takes care of that (See 
+[Using SingleModule](#using-singlemodule) below). 
+- `RouterModule` allows you to use the Play router without the rest of the configuration (see 
+[Using the router module directly](#using-the-router-module-directly).)
+
+#### Using `PlayModule`
+
+In order to use the `PlayModule` for your application, you need to provide the scala, Play and 
+Twirl versions. You also need to define your own test object which extends the provided 
+`PlayTests` trait. 
 
- 
 ```scala
 // build.sc
+import mill._
+// You have to replace VERSION
+import $ivy.`com.lihaoyi::mill-contrib-playlib:VERSION`,  mill.playlib._
+
+
+object core extends PlayModule {
+    //config
+    override def scalaVersion= T{"2.12.8"}
+    override def playVersion= T{"2.7.0"}
+    override def twirlVersion= T{"1.4.0"}
+    
+    object test extends PlayTests
+}  
+``` 
+
+Using the above definition, your build will be configured to use the default Play layout:
+```text
+.
+├── build.sc
+└── core
+    ├── app
+    │   ├── controllers
+    │   └── views
+    ├── conf
+    │   └── application.conf
+    │   └── routes
+    │   └── ...    
+    ├── logs
+    ├── public
+    │   ├── images
+    │   ├── javascripts
+    │   └── stylesheets
+    └── test
+        └── controllers
+```
+
+The following compile dependencies will automatically be added to your build:
```

Comment:
I think we should document the fact that we provide functions to add dependencies.
Maybe with an example.
Eg. if you want to add the `ws` module in your Play application:

```scala
def ivyDeps = super.ivyDeps() ++ Agg(ws())
```

Reply:
I have added a documentation section about this. I considered adding another one about overriding the static assets path but decided against it as it is definitely not something that's routinely done.